### ISSUE TITLE
unable to load package `bindings-GLFW-3.1.2.2'

### ISSUE THREAD

**Issue by PaulVisschers** [bsl/bindings-GLFW#48]
When I tried to install GLFW-b-1.4.8.1 through either Cabal or Stack, it will first install bindings-GLFW-3.1.2.2, but after succesfully doing so it moves on with GLFW-b and I get the following error:
```
ghc.exe: C:\Users\Paul\AppData\Roaming\cabal\x86_64-windows-ghc-8.0.1\bindings-G
LFW-3.1.2.2-14y8Rlh9wxcHUOlWbio9J9\HSbindings-GLFW-3.1.2.2-14y8Rlh9wxcHUOlWbio9J
9.o: unknown symbol `strdup'

ghc.exe: unable to load package `bindings-GLFW-3.1.2.2'
cabal: Leaving directory 'C:\Users\Paul\AppData\Local\Temp\cabal-tmp-1920\GLFW-b
-1.4.8.1'
Failed to install GLFW-b-1.4.8.1
cabal: Error: some packages failed to install:
GLFW-b-1.4.8.1 failed during the building phase. The exception was:
ExitFailure 1
```
Since the problem seems to be after the package is installed, this might be a bug in GHC or Cabal. I figured I'd report it here though, as you're probably more qualified to figure it out.

I worked around this by manually installing bindings-GLFW-3.1.2.1 instead of 3.1.2.2, that one works just fine.

**Comment by PaulVisschers** [comment]
I just tested and it works fine on GHC-7.10.3, but not on GHC-8.0.1.

**Comment by PaulVisschers** [comment]
It works fine in 32-bit GHC, so the issues are 64-bit specific. See also my comments in [this issue](https://github.com/bsl/GLFW-b/issues/44).

**Comment by shockkolate** [comment]
I'm also having this issue and I can confirm that it's with GHC 8.0.1 64-bit.

**Comment by Mistuke** [comment]
With the new linker in GHC 8.0.1 we stopped exporting POSIX functions under their deprecated names, this to be more inline with the most popular compiler on the platform (msvc and icc). The correct solution would be for `GLFW` to always on windows `#define strdup _strdup`. 

However I have introduced a change in the linker which will do this indirection if it's not already being done by the source.  This will be in GHC 8.0.2 https://ghc.haskell.org/trac/ghc/ticket/12497
